### DISCUSSION TITLE
Looping Connect-PnpOnline and MFA

### DISCUSSION THREAD

**Discussion by SP-Support** [pnp/powershell#486] [Q&A]
Since the new PNP.PowerShell looping through connecting to sites have gotten more difficult with MFA.

If you use `get-pnptenantsite` to get a list of sites and then need to connect to each site to get web details you'd use something like `connect-pnponline -interactive -url $site.url`

when you do this it asks for credentials for every site. However, using "-useweblogin" it gives a warning note to us -interact.

What is the proper way to loop through connections now?

**Comment by BaronSparky** [comment]
App registration identity? or an account that doesn't have MFA applied :(

**Reply by SP-Support** [reply]
Thanks for the thoughts but it seems that Connect-PnPOnline has gone through a change that causes it to ask for credentials on ever site in the loop. This is not the same behavior as it was before the upgrade.
